Marlin/src/pins/stm32f4/pins_BTT_OCTOPUS_V1_common.h

```diff
@@ -62,6 +62,21 @@
   #define Z_MIN_PROBE_PIN                   PB7
 #endif
 
+// Check for additional used endstop pins
+//
+#if ANY(X_DUAL_ENDSTOPS,Y_DUAL_ENDSTOPS,Z_MULTI_ENDSTOPS)
+  #define _ENDSTOP_IS_ANY(ES) X2_USE_ENDSTOP == ES || Y2_USE_ENDSTOP == ES || Z2_USE_ENDSTOP == ES || Z3_USE_ENDSTOP == ES || Z4_USE_ENDSTOP == ES
+  #if _ENDSTOP_IS_ANY(_XMIN_) || _ENDSTOP_IS_ANY(_XMAX_)
+    #define NEEDS_X_MINMAX
+  #endif
+  #if _ENDSTOP_IS_ANY(_YMIN_) || _ENDSTOP_IS_ANY(_YMAX_)
+    #define NEEDS_Y_MINMAX
+  #endif
+  #if _ENDSTOP_IS_ANY(_ZMIN_) || _ENDSTOP_IS_ANY(_ZMAX_)
+    #define NEEDS_Z_MINMAX
+  #endif
+#endif
+
```

Comment:
I imagine this check would be useful earlier / in the core of Marlin since we haven't seen the last of these "big" boards.

Reply:
Although, this way is not the only way to go. The chosen endstop is eventually based on whether homing direction is -1 or +1, and then the "`STOP`" pin will do for either one. The specific result below can work for this situation (where apparently the max should be different, if possible) but it may not be the best choice for all, so it kind of depends on the typical configurations they are expected to be deployed in.